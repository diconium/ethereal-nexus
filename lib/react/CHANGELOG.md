# @ethereal-nexus/react

## 0.2.0

### Minor Changes

- 90cf802: Update dependencies

## 0.1.2

### Patch Changes

- 205469c: Changed slot to allow the prop drill to nexus components inside experience fragments

## 0.1.1

### Patch Changes

- f882265: Check for cq-Editable-dom--container instead of section to spread the properties into a slot to fix issues when the page is not reloaded

## 0.1.0

### Minor Changes

- a367679: feat: add Slot component that enables drilling props to slotted nexus components
