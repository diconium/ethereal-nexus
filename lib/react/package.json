{
  "name": "@ethereal-nexus/react",
  "version": "0.1.1",
  "description": "",
  "keywords": [
    "react"
  ],
  "author": "",
  "license": "Apache-2.0",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "build": "tsup ./src",
    "dev": "tsup ./src --watch"
  },
  "types": "./dist/index.d.ts",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "require": "./dist/index.js",
      "import": "./dist/index.mjs"
    }
  },
  "sideEffects": false,
  "files": [
    "dist",
    "README.md"
  ],
  "devDependencies": {
    "@types/react": "^19.0.8",
    "tsup": "^8.0.2",
    "typescript": "^5.6.2"
  },
  "peerDependencies": {
    "react": "^18.0.0 || ^19.0.0",
    "react-dom": "^18.0.0 || ^19.0.0"
  }
}
