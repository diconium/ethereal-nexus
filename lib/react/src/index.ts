export { Slot } from './Slot/Slot';