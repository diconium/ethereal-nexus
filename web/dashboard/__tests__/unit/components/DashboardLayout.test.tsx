import React from 'react';
import { render, screen, act } from '@testing-library/react';
import DashboardLayout from '@/components/layout';
import { useTheme } from 'next-themes';

jest.mock('next-themes', () => ({
  ...jest.requireActual('next-themes'),
  useTheme: jest.fn().mockImplementation(() => ({
    resolvedTheme: undefined,
  })),
}));

jest.mock('@/components/user/user-nav', () => ({
  __esModule: true,
  UserNav: jest.fn().mockImplementation(() => <div>User Nav</div>),
}));

jest.mock('@/components/ui/toaster', () => ({
  __esModule: true,
  Toaster: jest.fn().mockImplementation(() => <div>Toaster</div>),
}));

jest.mock('@/components/ui/logo-image', () => ({
  __esModule: true,
  default: jest.fn().mockImplementation(() => <div>Ethereal Nexus Logo</div>),
}));

jest.mock('@/components/theme-picker', () => ({
  __esModule: true,
  default: jest.fn().mockImplementation(() => <div>ThemePicker</div>),
}));
jest.mock('@/components/ui/main-nav/main-nav', () => ({
  __esModule: true,
  MainNav: jest.fn().mockImplementation(() => <div>Main Nav</div>),
}));

// Set up mock implementation for window.matchMedia
window.matchMedia = jest.fn().mockImplementation(query => {
  return {
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    dispatchEvent: jest.fn(),
  };
});


describe('Dashboard layout component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <DashboardLayout>
        <div>Child Component</div>
      </DashboardLayout>
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it('should render loader if theme is undefined',  () => {
    render(
      <DashboardLayout>
        <div>Child Component</div>
      </DashboardLayout>
    );

    expect(screen.getByTestId('global-loader')).toBeInTheDocument();
    expect(screen.queryByText('Ethereal Nexus Logo')).not.toBeInTheDocument();
    expect(screen.queryByText('Main Nav')).not.toBeInTheDocument();
  });
  it('should render all child components if theme is defined',  () => {
    jest.useFakeTimers();
    jest.spyOn(global, 'setTimeout');

    (useTheme as jest.Mock).mockReturnValue({
      resolvedTheme: 'dark',
      setTheme: jest.fn(),
    });

    render(
      <DashboardLayout>
        <div>Child Component</div>
      </DashboardLayout>
    );

    expect(setTimeout).toHaveBeenCalledTimes(1);

    act(() => {
      jest.runOnlyPendingTimers();
    });

    expect(screen.queryByTestId('global-loader')).not.toBeInTheDocument();
    expect(screen.getByText('Ethereal Nexus Logo')).toBeInTheDocument();
    expect(screen.getByText('Main Nav')).toBeInTheDocument();
    expect(screen.getByText('ThemePicker')).toBeInTheDocument();
    expect(screen.getByText('User Nav')).toBeInTheDocument();
    expect(screen.getByText('Toaster')).toBeInTheDocument();
    expect(screen.getByText('Child Component')).toBeInTheDocument();
  });
});
