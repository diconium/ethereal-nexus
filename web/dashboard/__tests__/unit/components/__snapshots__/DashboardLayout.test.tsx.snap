// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard layout component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-col md:flex mt-6"
  >
    <div
      class="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2"
      data-testid="global-loader"
    >
      <span
        style="display: inherit; width: 100px; height: 100px; position: relative;"
      >
        <span
          style="position: absolute; top: 0px; left: 0px; width: 100px; height: 100px; border: 10px solid #ff5600; opacity: 0.4; border-radius: 100%; animation-fill-mode: forwards; animation: react-spinners-RingLoader-right 2s 0s infinite linear;"
        />
        <span
          style="position: absolute; top: 0px; left: 0px; width: 100px; height: 100px; border: 10px solid #ff5600; opacity: 0.4; border-radius: 100%; animation-fill-mode: forwards; animation: react-spinners-RingLoader-left 2s 0s infinite linear;"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;
